### ISSUE TITLE
Mysterious problem where labeled package in channel not picked up by conda-build

### ISSUE THREAD

**Issue by ezyang** [conda/conda-build#2448]
So, I have a kind of mysterious problem. I have a linux-64 machine, with conda 4.3.29 and conda-build 3.0.27 that exhibits the following symptoms:

1. When I say `conda build -c ezyang/label/gcc5 foo` where `foo` is a recipe directory that has a dependency on `protobuf`, the dependency solver for conda build always picks the defaults `protobuf` instead of picking up the gcc5 package in https://anaconda.org/ezyang/protobuf/files . If I pass `--override-channels` to prevent defaults from being picked, it doesn't manage to select `protobuf` at all.

2. However, when I go ahead and build protobuf using the conda-forge feedstock https://github.com/conda-forge/protobuf-feedstock and then do the build again, `conda build` is able to pick up the build and successfully do my build, IF I have `-c ezyang/label/gcc5` and I uploaded it to anaconda. In particular, if I OMIT `-c ezyang/label/gcc5`, my local tarball is not picked up at all.

3. If I upload this new protobuf to anaconda cloud with a label or without a label, it is still NOT picked up if I pass `-c ezyang/label/gcc5`.

So, basically, there is only one *specific* configuration where I can successfully get conda-build to use a particular protobuf build: (1) build it myself, (2) upload it to anaconda cloud, (3) set `-c` in conda build to point to that channel, and (4) have the local build product in my conda-bld/linux-64 directory.

To add insult to injury, I can't reproduce this problem on a different machine with the same versions of conda and conda-build.

Any ideas?

**Comment by msarahan** [comment]
I think the tricky part here may be unsatisfiable dependencies somehow.  You might be able to learn more about what conda is doing (and what dependencies it's missing) by doing something like this:

```conda create -n test_protobuf -c local -vvv protobuf```

**Comment by smithsp** [comment]
I am having a similar issue with conda-build 3.0.31 and conda 4.3.30.  I tried to modify the tk recipe from https://github.com/conda-forge/tk-feedstock to depend on `xorg-libx11` from the conda-forge channel using `conda-build -c conda-forge --override-channel tk-feedstock/recipe`
```
> conda-build -c conda-forge --override-channels tk-feedstock/recipe/
Attempting to finalize metadata for tk
INFO:conda_build.metadata:Attempting to finalize metadata for tk

Leaving build/test directories:
  Work:  /Users/smithsp/conda-omfit/conda-bld/work 
  Test:  /Users/smithsp/conda-omfit/conda-bld/test_tmp 
Leaving build/test environments:
  Test: source activate  /Users/smithsp/conda-omfit/conda-bld/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol 
  Build:        source activate  /Users/smithsp/conda-omfit/conda-bld/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p 


Traceback (most recent call last):
  File "/Users/smithsp/conda-omfit/bin/conda-build", line 11, in <module>
    sys.exit(main())
  File "/Users/smithsp/conda-omfit/lib/python2.7/site-packages/conda_build/cli/main_build.py", line 388, in main
    execute(sys.argv[1:])
  File "/Users/smithsp/conda-omfit/lib/python2.7/site-packages/conda_build/cli/main_build.py", line 379, in execute
    verify=args.verify)
  File "/Users/smithsp/conda-omfit/lib/python2.7/site-packages/conda_build/api.py", line 187, in build
    need_source_download=need_source_download, config=config, variants=variants)
  File "/Users/smithsp/conda-omfit/lib/python2.7/site-packages/conda_build/build.py", line 1801, in build_tree
    notest=notest,
  File "/Users/smithsp/conda-omfit/lib/python2.7/site-packages/conda_build/build.py", line 894, in build
    output_metas = expand_outputs([(m, need_source_download, need_reparse_in_env)])
  File "/Users/smithsp/conda-omfit/lib/python2.7/site-packages/conda_build/render.py", line 582, in expand_outputs
    for (output_dict, m) in _m.get_output_metadata_set(permit_unsatisfiable_variants=False):
  File "/Users/smithsp/conda-omfit/lib/python2.7/site-packages/conda_build/metadata.py", line 1799, in get_output_metadata_set
    permit_unsatisfiable_variants=permit_unsatisfiable_variants)
  File "/Users/smithsp/conda-omfit/lib/python2.7/site-packages/conda_build/metadata.py", line 658, in finalize_outputs_pass
    fm = finalize_metadata(om, permit_unsatisfiable_variants=permit_unsatisfiable_variants)
  File "/Users/smithsp/conda-omfit/lib/python2.7/site-packages/conda_build/render.py", line 314, in finalize_metadata
    permit_unsatisfiable_variants=permit_unsatisfiable_variants)
  File "/Users/smithsp/conda-omfit/lib/python2.7/site-packages/conda_build/render.py", line 115, in get_env_dependencies
    channel_urls=tuple(m.config.channel_urls))
  File "/Users/smithsp/conda-omfit/lib/python2.7/site-packages/conda_build/environ.py", line 693, in get_install_actions
    raise DependencyNeedsBuildingError(exc, subdir=subdir)
conda_build.exceptions.DependencyNeedsBuildingError: Unsatisfiable dependencies for platform osx-64: set(['xorg-libxll', 'libgcc'])
```

**Comment by smithsp** [comment]
Looking at using the branch at https://github.com/conda-forge/tk-feedstock/pull/17, I had not specified a version number, which they do, and building from that branch does seem to pick up the xorg-libx11 package OK.
